src/utils/asyncStore/identityStorage.js

```diff
@@ -143,8 +152,9 @@ export const updateIdentities = async (updatedIdentities) => {
 export const updateClaims = async (updatedClaims) => {
```

Comment:
You should not be doing these calculations in the storage repository. Same as storeSeedClaims.
The business logic needs to go into the sagas. 
These repository functions should be simple stupid, just retreiving and storing what you pass into them.